README.md

```diff
@@ -1 +1,188 @@
-# HoaHeap
+<img src="https://static.hoa-project.net/Image/Hoa.svg" alt="Hoa" width="280px" />
+
+[![Build status](https://img.shields.io/travis/hoaproject/heap/master.svg)](https://travis-ci.org/hoaproject/heap)
+[![Coverage](https://img.shields.io/coveralls/hoaproject/heap/master.svg)](https://coveralls.io/github/hoaproject/heap?branch=master)
+[![Packagist](https://img.shields.io/packagist/dt/hoa/heap.svg)](https://packagist.org/packages/hoa/heap)
+[![License](https://img.shields.io/packagist/l/hoa/heap.svg)](https://hoa-project.net/LICENSE)
+
+Hoa is a **modular**, **extensible** and **structured** set of PHP libraries.
+Moreover, Hoa aims at being a bridge between industrial and research worlds.
+
+# Hoa\Heap
+
+[![Help on IRC](https://img.shields.io/badge/help-%23hoaproject-ff0066.svg)](https://webchat.freenode.net/?channels=#hoaproject)
+[![Help on Gitter](https://img.shields.io/badge/help-gitter-ff0066.svg)](https://gitter.im/hoaproject/central)
+[![Documentation](https://img.shields.io/badge/documentation-hack_book-ff0066.svg)](https://hoa-project.net/Literature/Hack/heap.html)
+[![Board](https://img.shields.io/badge/organisation-board-ff0066.svg)](https://waffle.io/hoaproject/heap)
+
+This library provides a set of advanced Heap can support Scalar, Array, Object or Closure
+as item and not only Integer, as ordinal does.
+The order of heap depends of priority parameter.
+
+`Hoa\Heap\Min` and `Hoa\Heap\Max` class interpret priority by comparing items numerically.
+But you are free to implement your own class if you want a different sort algorithm.
+
+## :warning: Warning
+The default iteration process do not enqueue the Heap as common usage.
+You must use Generator methods `pop` or `pop` for iterate on with remove item from heap.
+
+## Installation
+
+With [Composer](http://getcomposer.org/), to include this library into your
+dependencies, you need to require
+[`hoa/heap`](https://packagist.org/packages/hoa/heap):
+
+```sh
+$ composer require hoa/heap '~0.0'
+```
+
+For more installation procedures, please read [the Source
+page](http://hoa-project.net/Source.html).
+
+## Testing
+
+Considering the library has been installed with Composer, the following
+commands will run the test suites:
+
+```sh
+$ composer install
+$ vendor/bin/hoa test:run
+```
+
+For more information, please consult the [contributor
+guide](https://hoa-project.net/Literature/Contributor/Guide.html).
+
+## Quick usage
+
```

Comment:
Quick usage use method as Heap value, but any kind of value can be specified, Scalar, Object, Resources...